sdk/Sdk/Tasks/GenerateFunctionMetadata.cs

```diff
@@ -60,6 +62,32 @@ public override bool Execute()
             return true;
         }
 
+        private void WriteMetadataWithRetry(IEnumerable<SdkFunctionMetadata> functions)
+        {
+            int attempt = 0;
+            while (attempt < 10)
+            {
+                try
+                {
+                    FunctionMetadataJsonWriter.WriteMetadata(functions, OutputPath!);
+                    break;
+                }
+                catch (IOException ex)
+                {
+                    attempt++;
+                    if (attempt == 10)
+                    {
+                        Log.LogErrorFromException(ex);
+                        throw new FunctionsMetadataGenerationException();
```

Comment:
Should we use the FunctionsMetadataGenerationException constructor which takes the message and inner exception?

Reply:
I can if you want - however it will have no affect.

1. I already log the exception on line 80. `Log.LogErrorFromException(ex);`.
2. Our catch block lines 56 to 62 make no use of the inner exception or message.

Reply:
I'll leave it for now. If/when we adjust the catch block then we can add it as an inner exception. I don't want to add it now, have the catch block be changed, and we end up double logging the inner error message (because the line 80 is left in also)